pkg/operator/ceph/object/controller.go

```diff
@@ -452,10 +453,14 @@ func (r *ReconcileCephObjectStore) reconcileCreateObjectStore(cephObjectStore *c
 		// Reconcile Multisite Creation
 		logger.Infof("setting multisite settings for object store %q", cephObjectStore.Name)
 		err = setMultisite(objContext, cephObjectStore, zone)
-		if err != nil && kerrors.IsNotFound(err) {
-			return reconcile.Result{}, err
-		} else if err != nil {
-			return r.setFailedStatus(k8sutil.ObservedGenerationNotAvailable, namespacedName, "failed to configure multisite for object store", err)
+		if err != nil {
+			if kerrors.IsNotFound(err) {
+				return reconcile.Result{}, err
+			} else if _e, ok := err.(kexec.CodeExitError); ok && _e.ExitStatus() == int(syscall.ENOENT) {
+				return reconcile.Result{}, err
```

Comment:
What does the ENOENT indicate? I'm having a hard time determining if we should use `r.setFailedStatus()` here and if we should include more info in the `err` by using `errors.Wrapf(err, ...)`.

From what I understand reading #12833 , this error is returned when Rook has successfully initiated the command in the mgr container and the command itself has failed. This is a different condition that when the mgr container isn't present, making it impossible to initiate the command.

Generally, I would expect that methods should always add context to `err` to help with debugging when something is broken. Unless there is a good reason not to, please use `errors.Wrapf(err, ...)` to add context here to aid in debugging.

Also, since the current behavior is to always use `r.setFailedStatus()` except for the single case where the error is a kubernetes IsNotFound error, I think this change should also do so, unless there is good reason not to.

Reply:
Turns out, these actually aren't necessary and are spill-over I have managed to miss when reducing redundancies.

For clarity, since they were conditioned on outputs of `RunAdminCommandNoMultisite` calls, they were also returning `kexec.CodeExitError`s, but which don't impact logic flow, unlike lines like https://github.com/rook/rook/blob/fc43f8fc691e3a389705365d29f1e9c816167227/pkg/operator/ceph/object/realm/controller.go#L245 scattered across `pkg/operator/ceph/object/{realm,zonegroup,zone}/controller.go`, which correctly returned `syscall.ENOENT` on `radosgw-admin realm/zonegroup/zone get` calls for not-yet present resources and worked properly on host-based networking, as these calls threw os/exec.ExitError, while mgr-proxied calls returned K8S' kexec.CodeExitError, not matching in https://github.com/rook/rook/blob/fc43f8fc691e3a389705365d29f1e9c816167227/pkg/util/exec/error.go#L37-L52 and failing correct reconciliation.